Modules/ITS/src/ITSThresholdCalibrationTask.cxx

```diff
@@ -0,0 +1,456 @@
+    hCalibrationThrNoiseLayer[iLayer]->SetStats(0);
+    formatAxes(hCalibrationThrNoiseLayer[iLayer], "THR noise (e) ", "Chip counts", 1, 1.10);
+    addObject(hCalibrationThrNoiseLayer[iLayer]);
+
+    hCalibrationThrNoiseRMSLayer[iLayer] = new TH1F(Form("ThrNoiseRMSLayer%d", iLayer), Form("ThrNoiseRMSLayer%d", iLayer), 50, -0.5, 9.5);
+    hCalibrationThrNoiseRMSLayer[iLayer]->SetStats(0);
+    formatAxes(hCalibrationThrNoiseRMSLayer[iLayer], "THR noise RMS (e)", "Chip counts", 1, 1.10);
+    addObject(hCalibrationThrNoiseRMSLayer[iLayer]);
+  }
+
+  hSuccessRate = new TH1F("SuccessRate", "Success Rate", 7, -0.5, 6.5);
+  hSuccessRate->SetStats(0);
+  formatAxes(hSuccessRate, "Layer", "Rate", 1, 1.10);
+  addObject(hSuccessRate);
+
+  for (int iLayer; iLayer < 7; iLayer++) {
+    SuccessStatus[iLayer] = 0;
+    TotalStatus[iLayer] = 0;
+  }
+}
+void ITSThresholdCalibrationTask::addObject(TObject* aObject)
+{
+  if (!aObject) {
+    ILOG(Info, Support) << " ERROR: trying to add non-existent object " << ENDM;
+    return;
+  } else {
+    mPublishedObjects.push_back(aObject);
+  }
+}
+
+void ITSThresholdCalibrationTask::formatLayers(TH2* h, Int_t iBarrel)
+{
+  Int_t iLayerBegin, iLayerEnd;
+
+  switch (iBarrel) {
+    case 0: {
+      iLayerBegin = 1;
+      iLayerEnd = 3;
+      break;
+    }
+    case 1: {
+      iLayerBegin = 4;
+      iLayerEnd = 5;
+      break;
+    }
+    case 2: {
+      iLayerBegin = 6;
+      iLayerEnd = 7;
+      break;
+    }
+  };
+
+  for (Int_t iLayer = iLayerBegin; iLayer <= iLayerEnd; iLayer++) {
+    TLatex* msg = new TLatex(0.5 + iBarrel * 4, StaveBoundary[iLayer - 1] + 2 + iBarrel * 2, Form("L%d", iLayer - 1));
+    msg->SetTextSize(18);
+    msg->SetTextFont(43);
+    h->GetListOfFunctions()->Add(msg);
+
+    if (iLayer < iLayerEnd) {
+      auto l = new TLine(-0.5, StaveBoundary[iLayer] - 0.5, nChipsPerStave[iLayer] - 0.5, StaveBoundary[iLayer] - 0.5);
+      h->GetListOfFunctions()->Add(l);
+    }
+  }
+}
+
+Int_t ITSThresholdCalibrationTask::getBarrel(Int_t iLayer)
+{
+
+  Int_t iBarrel;
+  if (iLayer < 3)
+    iBarrel = 0;
+  else if (iLayer < 5)
+    iBarrel = 1;
+  else
+    iBarrel = 2;
+
+  return iBarrel;
+}
+
+void ITSThresholdCalibrationTask::formatAxes(TH1* h, const char* xTitle, const char* yTitle, float xOffset, float yOffset)
+{
+  h->GetXaxis()->SetTitle(xTitle);
+  h->GetYaxis()->SetTitle(yTitle);
+  h->GetXaxis()->SetTitleOffset(xOffset);
+  h->GetYaxis()->SetTitleOffset(yOffset);
+}
+
+void ITSThresholdCalibrationTask::publishHistos()
+{
+  for (unsigned int iObj = 0; iObj < mPublishedObjects.size(); iObj++) {
+    getObjectsManager()->startPublishing(mPublishedObjects.at(iObj));
+    ILOG(Info, Support) << " Object will be published: " << mPublishedObjects.at(iObj)->GetName() << ENDM;
+  }
+}
+
+std::vector<std::string> ITSThresholdCalibrationTask::split(std::string s, std::string delimiter)
```

Comment:
always according to coding convention I would rename this function as splitString or something similar.